src/Cli/ConfigGenerator.cs

```diff
@@ -206,7 +211,7 @@ public static bool TryCreateRuntimeConfig(InitOptions options, FileSystemRuntime
                 Schema: dabSchemaLink,
                 DataSource: dataSource,
                 Runtime: new(
-                    Rest: new(!restDisabled, restPath ?? RestRuntimeOptions.DEFAULT_PATH),
+                    Rest: new(!restDisabled, restPath ?? RestRuntimeOptions.DEFAULT_PATH, options.RestRequestBodyStrict is CliBoolean.False ? false : true),
```

Comment:
If we set the default value as CliBoolean.True, we would not know the difference between whether this true value is because of default behavior or because of customer actually including this option in the init command. This might not seem very important here, but with this PR: https://github.com/Azure/data-api-builder/pull/1686, it will become very important where we would want to support negative/positive flags together and it will become imperative for us to know whether the flag was included in the init command or not.

Reply:
Sorry, I still dont understand the usage of CliBoolean.None. Why do we need to know if the flag was included in the command or not?
Why can't we set the property `Default = CliBool.True` for this option RestEnabled? 

This way even if the option is not specified in the command line it will be set the CliBool.True value - note its ok  if the Enum value is 1. 

See e.g. host-mode. If you dont specify this option, it will pick up the Production enum value (its numeric value is 1) - this happens because of the property Default on the option itself. it is not set to the Enum value 0. 
I looked at the PR #1686, even there why do we need CliBool.None?

Reply:
Assume a scenario where the init command looks like:

dab init ... --rest.disabled

which intends to disable the REST endpoint.


Now since the init command does not include the rest.enabled option, in the code, it would assume its default value.This default value can be CliBoolean.True/CliBoolean.None.


If we keep the default value as true, we are going to throw an exception because on one hand rest.disabled flag says that the rest endpoint is disabled while since the rest.enabled flag was not included, it indicates rest endpoint is enabled. However, we should be ignoring rest.enabled flag altogether because it was not even included in the init command. In this particular case, only rest.disabled should have been considered. Assigning a default value of CliBoolean.None helps us to identify that the option rest.enabled was not included in the init command and we can ignore that.

Reply:
Discussed over chat.